internal/app/secretless/handlers/mysql/backend.go

```diff
@@ -70,273 +118,310 @@ func (h *Handler) ConfigureBackend() (err error) {
 		result.Options[k] = string(v)
 	}
 
-	h.BackendConfig = &result
+	return result, nil
+}
+
+// https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_connection_phase.html
+//
+type ConnectionPhase struct {
+	connectionDetails *ConnectionDetails
+	clientConn        *MySQLConnection
+	backendConn       *MySQLConnection
+
+	sslMode *ssl.DbSSLMode
+
+	//TODO: after kumbi's work these 2 should be combined
+	rawServerHandshake      mysqlPacket
+	serverHandshake         *protocol.HandshakeV10
+	clientHandshakeResponse *protocol.HandshakeResponse41
 
-	return
+	err               error
 }
 
+func NewConnectionPhase(
+	clientConn *MySQLConnection,
+	backendConn *MySQLConnection,
+	connDetails *ConnectionDetails,
+) ConnectionPhase {
+	return ConnectionPhase{
+		connectionDetails: connDetails,
+		clientConn:        clientConn,
+		backendConn:       backendConn,
+	}
+}
 
-// ConnectToBackend establishes the connection to the backend database and sets the Backend field.
-func (h *Handler) ConnectToBackend() (err error) {
+func (mc *ConnectionPhase) Run() error {
+	mc.readServerHandshake()
+	mc.writeHandshakeToClient()
+	mc.validateServerSSL()
+	mc.readClientHandshakeResponse()
+	mc.overrideClientCapabilities()
+	mc.injectCredentials()
+	mc.maybeUpgradeToSSL()
+	mc.writeClientHandshakeResponseToBackend()
```

Comment:
the sketch at the top of backend.go should be updated to reflect this flow - there seems to be a slight mismatch right now